src/components/dialogs/apiKeySelections.ts

```diff
@@ -0,0 +1,23 @@
+export function getStoreApiKeys() {
+  try {
+    if (localStorage.getItem('store-api')) {
+      return JSON.parse(localStorage.getItem('store-api')!) as boolean;
+    }
+    return false;
+  } catch {
+    localStorage.removeItem('store-api');
+    return false;
+  }
```

Comment:
a few things:
* lets call these `'store-api-key'` and `'api-key'` because "api" by itself could mean any number of different things.
* should this code be in `src/store.ts` where the other localStorage stuff is done (or vice-versa)?
* do we even need the JSON parsing?  you dont appear to be encoding the value when it is stored in `src/store.ts`...  it seems that my api key value is not being preserved between visits to the netlify preview, and i think this is why (though i havent really explored the reason).
* is this suggestion simpler while still making sense?  it makes the `try` tighter, removes the need for the `catch`, and deduplicates the default value:

```suggestion
  if (localStorage.getItem('store-api')) {
    try {
      // if we saved it, return it
      return JSON.parse(localStorage.getItem('store-api')!) as boolean;
    }
    // if parsing fails, saved value is bad, so clear it out
    localStorage.removeItem('store-api');
  }
  // if none of the above worked, return default of 'false'
  return false;
```

either way, it seems like kind of a lot of work to return a stored value...  if this truly is a common pattern, you could do something like:
```ts
// helper function that provides common 'getter' boilerplate code
export function buildGetter(keyName, defaultValue, returnType) {
  // TODO: dont require returnType arg, instead using something like:
  //       `returnType = typeof defaultValue`
  //       (which wont work because typeof returns a string)
  if (localStorage.getItem(keyName)) {
    try {
      // if we saved it, return it
      return JSON.parse(localStorage.getItem(keyName)!) as returnType;
    }
    // if parsing fails, saved value is bad, so clear it out
    localStorage.removeItem(keyName);
  }
  // if none of the above worked, return default
  return defaultValue;
}

export function getStoreApiKeys() {
  return buildGetter('store-api', false, boolean);
}

export function getApiKey() {
  return buildGetter('api', '', string);
}
```

Reply:
Fixed up the first four bullet points!

As for the generic getter, in JavaScript every non-empty string is "truthy", and it's not clever enough to recognize that the string "false" is the boolean "false". So the type conversion actually has to be done like this:
```
localStorage.getItem('store-api-key') === 'true'
```

Reply:
> every non-empty string is "truthy", and it's not clever enough to recognize that the string "false" is the boolean "false"

perhaps...  but you might notice that the following evaluates to the boolean `true`:
```js
JSON.parse('false') == false
```